reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -144,23 +143,39 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
     }
   }
 
-  private def storeSignedTarget[T](repoId: RepoId, signed: SignedPayload[TargetsRole]): Route = {
-    val f: Future[ToResponseMarshallable] = offlineSignedRoleStorage.store(repoId, signed).map {
-      case Valid(_) =>
-        StatusCodes.NoContent
+  private def storeSignedTarget[T](repoId: RepoId, namespace: Namespace, signed: SignedPayload[TargetsRole]): Route = {
+    onSuccess(offlineSignedRoleStorage.store(repoId, signed)) {
+      case Valid(newSignedRole) =>
+        newSignedRole.content.signed.as[TargetsRole].right.foreach { targetRole =>
+          publishNewTargets(namespace, targetRole.targets)
+        }
+        respondWithCheckSum(newSignedRole.checksum.hash) {
+          complete(StatusCodes.NoContent)
+        }
       case Invalid(errors) =>
         val obj = Json.obj("errors" -> errors.asJson)
-        StatusCodes.BadRequest -> obj
+        complete(StatusCodes.BadRequest -> obj)
     }
-
-    complete(f)
   }
 
-  private def updateSignedTarget(repoId: RepoId, signedPayload: SignedPayload[TargetsRole], existentRole: SignedRole) =
+  private def publishNewTargets(namespace: Namespace, targets: Map[TargetFilename, ClientTargetItem]): Unit =
+    targets.foreach { case (targetFilename, clientTargetItem) =>
+      clientTargetItem.hashes.headOption.foreach { case (hashMethod, validChecksum) =>
+        messageBusPublisher.publish(TufTargetAdded(namespace, targetFilename, Checksum(hashMethod, validChecksum),
+                                            clientTargetItem.length, clientTargetItem.customParsed[TargetCustom]))
+      }
+    }
```

Comment:
So this will publish all signed targets, even targets that already existed, is this ok ?

Reply:
It gets called in `storeSignedTarget` after no existing target role has been found or in `updateSignedTarget` with the new targets minus the existing ones.

Reply:
Ok got confused with the name `publishNewTargets`. 

I think this is too much logic to have on Resource, these things should probably be in a separate class where it can be unit tested, we should not need to test the resource to test if targets are added or not.

`OfflineSignedRoleStorage` should receive a bus, to publish the targets, or some class that wraps the bus, such class could also be shared to wherever else we need to trigger those events.

The resource could even just call `OfflineSignedRoleStorage` and then that wrapper class with the resulting targets, and all this logic would be contained in that class, I think that would be a step in the right direction.